### PR TITLE
Feature/frontend api

### PR DESCRIPTION
**Pull request by moonryc**
Added MUI, grapghql,  added forms, set up useMutations for logging in and signingup wwhich DO NOT CURRENTLY WORK.

### CODE REVIEW

File: client/src/components/LoginForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  const handleFormSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    if(!email || !password){
+      return
+    }
+
+    try{
+      //TODO
```
Comment (jollypong) [inline client/src/components/LoginForm/index.tsx:38]: I thought TODO's wern't supposed to be pushed?

Comment (moonryc) [reply client/src/components/LoginForm/index.tsx:38]: This is just a todo as a reminder to fix the //@ts-ignore

File: client/src/components/LoginForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    setLoginForm((prevState) => ({
+      ...prevState,
+      [e.target.name]: e.target.value,
+    }));
+  };
+
+  const handleFormSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    if(!email || !password){
```
Comment (jollypong) [inline client/src/components/LoginForm/index.tsx:33]: Should add error message if email or password not there

File: client/src/components/NavBar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { auth } from '../../utils/auth';
+
+export const NavBar = () => {
```
Comment (jollypong) [inline client/src/components/NavBar/index.tsx:5]: I thought Travis said no NavBar for this project?

Comment (moonryc) [reply client/src/components/NavBar/index.tsx:5]: you are absolutly correct, I wasnt thinking,  I'll remove it either if this pull request is denied or approved

File: client/src/utils/auth.ts [hunk 4]
```diff
@@ -1,20 +1,27 @@
 // use this to decode a token and get the user's information out of it
 import decode from 'jwt-decode';
 
+export const auth={
 // get user data
-export const getProfile = () => {
-  return decode(getToken() as string);
-};
+getProfile:() => {
```
Comment (jollypong) [inline client/src/utils/auth.ts:6]: We don't have a profile page i thought?

Comment (moonryc) [reply client/src/utils/auth.ts:6]: thanks I will remove this either after the pull request is approved or when it is rejected, this is from template code

File: client/src/utils/auth.ts [hunk 5]
```diff
@@ -24,22 +31,21 @@ export const isTokenExpired = (token: string) => {
...
   // Saves user token to localStorage
   localStorage.setItem('id_token', idToken);
-  window.location.assign('/');
-};
+},
 
-export const logout = () => {
+ logout : () => {
   // Clear user token and profile data from localStorage
   localStorage.removeItem('id_token');
```
Comment (jollypong) [inline client/src/utils/auth.ts:48]: we don't need to route them to login page after logout?

**Review by jollypong — CHANGES_REQUESTED** [review]
Added some comments on possible changes. 
ps. ur ts-ignore isn't working!

**Review by jollypong — APPROVED** [review]
LGTM!

**Pull request merged**